Automattic/jetpack
# Unify WordPress Posts Widget with WP.com's version

#### Changes proposed in this Pull Request:

* Split the WordPress Posts Widget class into two to make it easier to reconcile WordPress.com v. Jetpack code/behavior.
* I think the only functional change is that `Jetpack_Display_Posts_Widget::update_instance()` now returns the newly updated data instead of `void`.

Note: For backward compatibility, the final class (in this case, the outermost class in the inheritance tree) must:
* be called `Jetpack_Display_Posts_Widget` for backward compatibility reasons and
* have a constructor that accepts no arguments.

#### Testing instructions:

`$ phpunit --filter WP_Test_Jetpack_Display_Posts_Widget`

A simple sanity check in addition to the automated tests above:

1. Before patching, add a WordPress Posts widget to a sidebar in your theme.
2. Apply patch.
3. Make sure the Widget still displays correctly.
4. Wait 11 minutes :)
5. Make sure the Widget still displays correctly.
6. Add a new WordPress Posts widget (pulling data from a different site) to a sidebar in your theme.
7. Make sure both widgets display correctly.
8. Edit the first widget to pull data from a different site.
9. Make sure everything displays correctly.

Corresponding WordPress.com patch: D13725
